src/os/filestore/HashIndex.cc

```diff
@@ -319,9 +319,13 @@ int HashIndex::split_dirs(const vector<string> &path, int target_level) {
   }
 
   if (must_split(info, target_level)) {
+    const coll_t c = coll();
+    spg_t pg;
+    if (!c.is_pg_prefix(&pg))
+      return -EINVAL;
```

Comment:
I don't think this should happen, but if the meta collection somehow does split, let's not return an error, just output "meta" instead of the pgid - an EINVAL here won't be handled well by the upper layers

Reply:
@jdurgin thank you Josh for your inputs. I misunderstood it. If I understand meta here means the current/meta directory which holds all the osdmap and other metadata stuff. If my understanding is correct we have never seen that gets split I think then better to drop this change and only keep _created function one.

Reply:
the change makes sense for both functions, we should just avoid returning EINVAL for both. just ignoring the return value of is_pg_prefix() in both places is fine.